lib/chef/knife/ec2_server_create.rb

```diff
@@ -851,6 +851,52 @@ def eip_scope
         end
       end
 
+      def ssl_config_user_data
+<<-EOH
+
+$vm_name = invoke-restmethod -uri http://169.254.169.254/latest/meta-data/public-ipv4
+winrm quickconfig -q
+New-SelfSignedCertificate -certstorelocation cert:\\localmachine\\my -dnsname $vm_name
+$thumbprint = (Get-ChildItem -Path cert:\\localmachine\\my | Where-Object {$_.Subject -match "$vm_name"}).Thumbprint;
+$create_listener_cmd = "winrm create winrm/config/Listener?Address=*+Transport=HTTPS '@{Hostname=`"$vm_name`";CertificateThumbprint=`"$thumbprint`"}'"
+iex $create_listener_cmd
+
+netsh advfirewall firewall add rule name="WinRM HTTPS" protocol=TCP dir=in Localport=5986 remoteport=any action=allow localip=any remoteip=any profile=public enable=yes
+
+EOH
+      end
+
+      def ssl_config_data_already_exist?
+        File.read(locate_config_value(:aws_user_data)).gsub(/\\\\/,"\\").include? ssl_config_user_data
+      end
+
+      def modify_user_data_file(contents)
+        File.open(locate_config_value(:aws_user_data),'w') do |file|
+          file.write contents
+        end
+      end
+
+      def process_user_data
+        if !ssl_config_data_already_exist?
+          ps_start_tag = "<powershell>\n"
+          ps_end_tag = "</powershell>\n"
+          script_lines = File.readlines(locate_config_value(:aws_user_data))
+          ps_start_tag_index = script_lines.index(ps_start_tag) || script_lines.index(ps_start_tag.strip)
+          ps_end_tag_index = script_lines.index(ps_end_tag) || script_lines.index(ps_end_tag.strip)
+          case
+          when ( ps_start_tag_index && !ps_end_tag_index ) || ( !ps_start_tag_index && ps_end_tag_index )
+            ui.error("Provided user_data file is invalid.")
```

Comment:
I think this will break following valid use case?
user-data file contains - 
```
<script>
echo \"%USERNAME%\" > 
.
.
</script>
```

Also consider below combo- 
```
<powershell> 
.
.
</powershell>
<script>
.
.
</script>
```